src/diffpy/utils/parsers/serialization.py

```diff
@@ -64,64 +64,69 @@ def serialize_data(
         Returns the dictionary loaded from/into the updated database file.
     """
 
-    # compile data_table and hddata together
+    # Combine data_table and hdata together in data
     data = {}
 
-    # handle getting name of file for variety of filename types
-    abs_path = pathlib.Path(filename).resolve()
-    # add path to start of data if requested
+    # Handle getting name of file for variety of filename types
+    abs_path = Path(filepath).resolve()
+
+    # Add path to start of data if show_path is True
     if show_path and "path" not in hdata.keys():
         data.update({"path": abs_path.as_posix()})
-    # title the entry with name of file (taken from end of path)
-    title = abs_path.name
 
-    # first add data in hddata dict
+    # Add hdata to data
     data.update(hdata)
 
-    # second add named columns in dt_cols
-    # performed second to prioritize overwriting hdata entries with data_table column entries
-    named_columns = 0  # initial value
-    max_columns = 1  # higher than named_columns to trigger 'data table' entry
-    if dt_colnames is not None:
-        num_columns = [len(row) for row in data_table]
-        max_columns = max(num_columns)
-        num_col_names = len(dt_colnames)
-        if max_columns < num_col_names:  # assume numpy.loadtxt gives non-irregular array
-            raise ImproperSizeError("More entries in dt_colnames than columns in data_table.")
-        named_columns = 0
-        for idx in range(num_col_names):
-            colname = dt_colnames[idx]
-            if colname is not None:
-                if colname in hdata.keys():
+    # Prioritize overwriting hdata entries with data_table column entries
+    col_counter = 0
+
+    # Get a list of column counts in each entry in data table
+    dt_col_counts = [len(row) for row in data_table]
+    dt_max_col_count = max(dt_col_counts)
+
+    if dt_col_names is not None:
+        dt_col_names_count = len(dt_col_names)
+        if dt_max_col_count < dt_col_names_count:  # assume numpy.loadtxt gives non-irregular array
```

Comment:
Compare the number of `col_names` values provided as an input for this function against the number of col determined from the data table.